source/updateCheck.py

```diff
@@ -94,6 +97,66 @@
 autoChecker: Optional["AutoUpdateChecker"] = None
 
 
+@dataclass
+class UpdateInfo:
+	"""
+	Data class representing update information for NVDA.
+
+	Attributes:
+		version (str): The version of the update.
+		launcherUrl (str): The URL to download the launcher.
+		apiVersion (str): The version of the API.
+		launcherHash (Optional[str]): The hash of the launcher, if available.
+		apiCompatTo (Optional[str]): The API compatibility version, if available.
+		changesUrl (Optional[str]): The URL to the changelog, if available.
+		launcherInteractiveUrl (Optional[str]): The URL for the interactive launcher, if available.
+	"""
+
+	version: str
+	launcherUrl: str
+	apiVersion: str
+	launcherHash: Optional[str] = None
+	apiCompatTo: Optional[str] = None
+	changesUrl: Optional[str] = None
+	launcherInteractiveUrl: Optional[str] = None
+
+	@classmethod
+	def parseUpdateCheckResponse(cls, data: str) -> Optional["UpdateInfo"]:
+		"""
+		Parses the update response and returns an UpdateInfo object.
+
+		:param data: The raw server response as a UTF-8 decoded string.
+		:return: An UpdateInfo object containing the update metadata, or None if the format is invalid.
+		:raises ValueError: If the response format is invalid.
+		"""
+		required_keys = ["version", "launcherUrl", "apiVersion"]
+		metadata = {}
+		for line in data.splitlines():
+			try:
+				key, val = line.split(": ", 1)
+				metadata[key] = val
```

Comment:
If we use a similar approach to above, we can grab all good keys, and silently drop bad keys, so we only need to unpack `metadata` to initialise `cls`. We can also remove keys from `requiredKeys` to make checking for all required keys easier later.
```suggestion
				if key in recognizedKeys:
					metadata[key] = val
					requiredKeys.difference_update(key)
				else:
					log.debug(f"Dropping unknown key {key} = {val}.")
```